subprojects/provider/src/main/kotlin/org/gradle/kotlin/dsl/resolver/ResolverEventLogger.kt

```diff
@@ -104,6 +107,21 @@ object DefaultResolverEventLogger : ResolverEventLogger {
     private
     fun now() = GregorianCalendar.getInstance().time
 
+    private
+    fun cleanupLogDirectory(logDir: File) {
+        val expiration = GregorianCalendar.getInstance().apply { add(Calendar.DAY_OF_MONTH, -cleanupAfterDays) }.time
+        logDir.listFiles { file ->
+            file.isFile && file.name.matches(resolverLogFilenameRegex) && Date(file.lastModified()).before(expiration)
+        }.forEach { it.delete() }
+    }
```

Comment:
In some rare cases, you'll have a system with a misconfigured time server that's date starts on 1 January 1970. If you subtract 7 days from 1 January 1970, do you end up with an error, or do you end up with all the log files deleted because of a time-based underflow?

If you're not concerned, feel free to ignore this.

Reply:
Right, the time based underflow would have the effect of cleaning up all log files right before creating the current one. I'm not too concerned, if you are in January the 1st 1970 you have other fish to fry than troubleshoot IJ editing kotlin build logic.